sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -0,0 +1,266 @@
+  }
+
+  /**
+   * Reissues a new MSFAA number to be associated with the student, cancelling any
+   * pending MSFAA for the particular offering intensity and also associating the
+   * new MSFAA number to any pending disbursement for the same offering intensity.
+   * Reissuing an MSFAA is only valid for an application with pending disbursement
+   * where the MSFAA is cancelled.
+   * @param referenceApplicationId reference application id.
+   * @returns the newly created MSFAA.
+   */
+  async reissueMSFAA(
+    referenceApplicationId: number,
+    auditUserId: number,
+  ): Promise<MSFAANumber> {
+    const application = await this.getApplicationForMSFAACreation(
+      referenceApplicationId,
+    );
+    const pendingDisbursement =
+      application.currentAssessment.disbursementSchedules.find(
+        (schedule) =>
+          schedule.disbursementScheduleStatus ===
+          DisbursementScheduleStatus.Pending,
+      );
+    if (!pendingDisbursement) {
+      throw new CustomNamedError(
+        "Not possible to reissue an MSFAA when there is no pending disbursements for the application.",
+        APPLICATION_INVALID_DATA_TO_CREATE_MSFAA_ERROR,
+      );
+    }
+    if (!pendingDisbursement.msfaaNumber.cancelledDate) {
+      throw new CustomNamedError(
+        "Not possible to reissue an MSFAA when the current associated MSFAA is not cancelled.",
+        APPLICATION_INVALID_DATA_TO_CREATE_MSFAA_ERROR,
+      );
+    }
+    return this.internalCreateMSFAANumber(
+      application.student.id,
+      application.id,
+      application.currentAssessment.offering.offeringIntensity,
+      auditUserId,
+    );
+  }
+
+  /**
+   * Creates a new MSFAA record with a new number for the specified student.
+   * @param studentId student to have a new MSFAA record created.
+   * @param referenceApplicationId reference application id.
+   * @param offeringIntensity offering intensity.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * individually based on, for instance, the Part time/Full time.
+   * @returns created MSFAA record.
+   */
+  private async internalCreateMSFAANumber(
+    studentId: number,
+    referenceApplicationId: number,
+    offeringIntensity: OfferingIntensity,
+    auditUserId: number,
+  ): Promise<MSFAANumber> {
+    return this.dataSource.transaction(async (entityManager) => {
+      const auditUser = await this.systemUsersService.systemUser();
+      const now = new Date();
+      const nowISODate = getISODateOnlyString(now);
+      // Cancel any pending MSFAA record that was never reported as signed or cancelled.
+      // This will ensure that the newly created MSFAA will be the only one valid.
+      await entityManager.getRepository(MSFAANumber).update(
+        {
+          student: { id: studentId },
+          offeringIntensity,
+          dateSigned: IsNull(),
+          cancelledDate: IsNull(),
+        },
+        {
+          cancelledDate: nowISODate,
```

Comment:
mm, I was thinking do we need any other indication to know this process happened as part of reissuing the new MSAFAA and was not canceled as part of the file integration? or do we care about that?